pkg/resource/grafana-agent-secret/reconciler.go

```diff
@@ -0,0 +1,95 @@
+package grafanaagentsecret
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	apimachineryerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/giantswarm/logging-operator/pkg/common"
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	loggingcredentials "github.com/giantswarm/logging-operator/pkg/resource/logging-credentials"
+
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// Reconciler implements a reconciler.Interface to handle
+// grafana-agent secret: extra secret which stores logging write credentials
+type Reconciler struct {
+	client.Client
+}
+
+// ReconcileCreate ensures grafana-agent secret is created with the right credentials
+func (r *Reconciler) ReconcileCreate(ctx context.Context, lc loggedcluster.Interface) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+	logger.Info("grafana-agent-secret create")
+
+	// Retrieve secret containing credentials
+	var loggingCredentialsSecret v1.Secret
+	err := r.Client.Get(ctx, types.NamespacedName{Name: loggingcredentials.LoggingCredentialsSecretMeta(lc).Name, Namespace: loggingcredentials.LoggingCredentialsSecretMeta(lc).Namespace},
```

Comment:
I think you can use client.ObjectKeyFromObject{loggingcredentials.LoggingCredentialsSecretMeta(lc)} instead of the whole types.NamespacedName{Name: loggingcredentials.LoggingCredentialsSecretMeta(lc).Name, Namespace: loggingcredentials.LoggingCredentialsSecretMeta(lc).Namespace}

Reply:
I got error
```
cannot use loggingcredentials.LoggingCredentialsSecretMeta(lc) (value of type "k8s.io/apimachinery/pkg/apis/meta/v1".ObjectMeta) as client.Object value in argument to client.ObjectKeyFromObject: "k8s.io/apimachinery/pkg/apis/meta/v1".ObjectMeta does not implement client.Object (missing method DeepCopyObject)compiler[InvalidIfaceAssign](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#InvalidIfaceAssign)
```

Reply:
Alright